flink-streaming-java/src/test/java/org/apache/flink/streaming/api/operators/SourceOperatorAlignmentTest.java

```diff
@@ -167,6 +167,9 @@ public void testWatermarkAlignmentWithIdleness() throws Exception {
             assertThat(operator.emitNext(actualOutput), is(DataInputStatus.NOTHING_AVAILABLE));
             context.getTimeService().advance(1);
             assertLatestReportedWatermarkEvent(context, Long.MAX_VALUE);
+            // receive Long.MAX_VALUE as WatermarkAlignmentEvent
```

Comment:
I think you want to check the `WatermarkAlignmentEvent`? and then assert on whether the maxWatermark is correct

Reply:
Yes! I have read the code just now, I find the `currentMaxDesiredWatermark` is maxWatermark by default, that's why the old test passed. I'm confused whether it's right to write like this now.

Reply:
Oh, I see. Does it make sense to call operator.handleOperatorEvent(new WatermarkAlignmentEvent(...)) after every `assertLatestReportedWatermarkEvent` (line 162)?

Reply:
I think so, and I have fixed it.